### PR TITLE
feat: rule no-unnormalized-keys

### LINKED ISSUE #32: New rule: Normalize string keys
### Rule details

Normalize string keys

### What type of rule is this?

Suggests an alternate way of doing something

### Example code

Per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize , there are examples of unicode that are similar in character. Generally this could mean that there is a key mismatch on 2 apparently similar strings. Suggesting using a normalized form to avoid this potential issue would be good so that the following 2 object don't have mismatched keys:

```json
{
  "Amélie": 1
}
```

```json
{
  "Amélie": 2
}
```

There are valid use cases for these key mismatches, but they are generally very specific in scope.

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

Similar mismatches or deception such as using Bidi mixing / zero width characters etc. are potentially in the same vein but likely would be a different scope since those have associated CVEs.

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

-   [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?

Adds a new `no-unnormalized-keys` rule.

#### What changes did you make? (Give an overview)

Ported over from #38.

- Ensured that the rule is exported from the plugin
- Updated README
- Switched rule options to be an object with a named propert
- Updated tests

#### Related Issues

Fixes #32
Closes #38

### CODE REVIEW

**Comment by nzakas** [comment]
I think I got all of the outstanding work for this rule.

File: src/rules/no-unnormalized-keys.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		messages: {
+			unnormalizedKey: "Unnormalized key '{{key}}' found.",
+		},
+
+		schema: [
+			{
+				type: "object",
+				properties: {
+					form: {
+						type: "string",
```
Comment (mdjermanovic) [inline src/rules/no-unnormalized-keys.js:23]:
```suggestion
```

Unnecessary with `enum`.

File: tests/plugin/eslint.test.js
[the remark this answers is not in the document]
Comment (nzakas) [reply tests/plugin/eslint.test.js:1]: Oops yes

File: src/rules/no-unnormalized-keys.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+/**
+ * @fileoverview Rule to detect unnormalized keys in JSON.
+ * @author Bradley Meck Farias
+ */
+
+export default {
+	meta: {
+		type: "problem",
```
Comment (mdjermanovic) [inline src/rules/no-unnormalized-keys.js:8]:
```suggestion
		type: /** @type {const} */ ("problem"),
```

To fix the Test Types failure.

(the same change was done in other rules in https://github.com/eslint/json/pull/65).

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**